### PR TITLE
Fixed: cell properties dialog shows styles that are not allowed through allowedContent

### LINKED ISSUE #1986: Cell properties shows style elements that are not allowed through allowedContent
## Type of report

Bug

## Provide detailed reproduction steps (if any)

This is a followup to a previous issue https://github.com/ckeditor/ckeditor-dev/issues/1014

Which is based on an issue in Drupal: https://www.drupal.org/project/drupal/issues/2785483

1. Assume the following allowedContent: 
```
config.allowedContent = {
		td: {
			attributes: 'colspan rowspan',
			styles: false
		}
	}
```
2. Add a table and open the 'Cell properties' dialog

### Expected result

* The 'Cell properties' dialog shows form elements for 'Cell Type', 'Rows Span', 'Columns Span'

### Actual result

* The 'Cell properties' dialog still shows form elements for 'Word Wrap', 'Horizontal Alignment', 'Vertical Alignment', 'Background Color', 'Border Color'

## Other details

* Browser: All
* OS: All
* CKEditor version: -
* Installed CKEditor plugins: tabletools

### PR DESCRIPTION
**Pull request by engineering-this**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_contributing_code-section-tests),
[how to set the testing environment](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_tests) and
[how to create tests](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_tests-section-creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What changes did you make?

I've added `requiredContent` for each option in table cell properties dialog.

Closes #1986

### CODE REVIEW

**Comment by engineering-this** [comment]
- I have rebased it on current master branch.
- Reworked how dialog is rendered. I think it looks much better now.
- Added manual test that allows to check various configurations.

**Comment by engineering-this** [comment]
> - it's not possible to have dialog with only width or only height; it's possible only to have dialog with both of these settings;
As originally both width, and height inputs had `requiredContent: 'td{width,height}',`, but I guess you are right and it's ok to change it within this issue.

> - unit selection for height is broken.
➡️ #2084

File: tests/plugins/tabletools/manual/allowedcontent.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+@bender-tags: 4.10.1, 1986, bug
```
Comment (Comandeer) [inline tests/plugins/tabletools/manual/allowedcontent.md]: Please update version tag.

**Comment by engineering-this** [comment]
Rebasted onto major, updated version tag and reworked unit test.

**Comment by engineering-this** [comment]
Rebased and fixed failing test on IE8.

**Comment by engineering-this** [comment]
- Changed dialog, so items will split into two columns evenly whenever there is at least 5 items, otherwise they will be in one column. 
- Dynamically removing spacers if they are last items.
- Removed spacers after bgColor and borderColor, fields will take all free space unless color dialog plugin is loaded. If it does dialog will look same as before.
- Updated tests after changes.
- Rebased.

**Comment by engineering-this** [comment]
- Rebased onto latest major.
- Improved editor names in unit test, so each editor has both index and allowed rules included in name.
- Fixed color dialog in manual test.
- Added new spacer after alignment, which will only appear whenever last item from alignment group is not last in column.
- Fixed spacer after rowspan/colspan group, which appeared only if the second existed. Currently it appears whenever at least one of them is present.

Didn't mention it in comment before:

In previous push I've removed spacer between bgcolor and bordercolor, as they should be treated as one group.

**Comment by Comandeer** [comment]
Rebased onto latest `master`.

**Pull request merged**